parteh/PRTGenericMod.F90

```diff
@@ -0,0 +1,1196 @@
+  ! arbitrary hypothesis.
+  ! These are things that are generally true, not specific to each plant.
+  ! For instance the map just contains the list of variable names, not the values for
+  ! each plant.
+  ! These are not instanced on every plant, they are just instanced once on every model 
+  ! machine or memory space. They should only be initialized once and used
+  ! as read-only from that point on.
+  ! -------------------------------------------------------------------------------------
+
+  ! -------------------------------------------------------------------------------------
+  ! This type simply packs the names and symbols associated with all
+  ! the variables for any given hypothesis
+  ! -------------------------------------------------------------------------------------
+  
+  type :: state_descriptor_type
+     character(len=maxlen_varname)   :: longname
+     character(len=maxlen_varsymbol) :: symbol
+     integer                         :: organ_id    ! global id for organ
+     integer                         :: spec_id     ! global id for species
+     integer                         :: num_pos     ! number of descrete spatial positions
+
+     ! Also, will probably need flags to define different types of groups that this variable
+     ! belongs too, which will control things like fusion, normalization, when to zero, etc...
+
+  end type state_descriptor_type
+  
+
+
+  ! This type will help us loop through all the different variables associated
+  ! with a specific organ type. Since variables are a combination of organ and
+  ! species, the number of unique variables is capped at the number of species
+  ! per each organ.
+  
+  type organ_map_type
+     integer, dimension(1:num_species_types) :: var_id
+     integer                                 :: num_vars
+  end type organ_map_type
+
+
+  ! This structure packs both the mapping structure and the variable descriptors
+  ! -------------------------------------------------------------------------------------
+  ! This array should contain the lists of indices to 
+  ! the species x organ variable structure that is used to map variables to the outside
+  ! world.
+  !   
+  !              
+  !                   | carbon | nitrogen | phosphorous | .... |
+  !                   ------------------------------------------
+  !    leaf           |        |          |             |      |
+  !    fine-root      |        |          |             |      |
+  !    sapwood        |        |          |             |      |
+  !    storage        |        |          |             |      |
+  !    reproduction   |        |          |             |      |
+  !    structure      |        |          |             |      |
+  !    ....           |        |          |             |      |
+  !                   ------------------------------------------
+  !     
+  ! -------------------------------------------------------------------------------------
+
+
+  type prt_instance_type
```

Comment:
Add some descriptor like "type variables specific to each hypothesis (carbon allocation, etc.) In general these generic routines (by nature of their need to accommodate allcomers, hypothesis and structure-wise)might need a bit of commenting love to keep potential future developers from being overwhelmed.